pkg/user/appointee/list.go

```diff
@@ -1,25 +1,158 @@
 package appointee
 
 import (
+	"context"
+	"fmt"
+	"github.com/Layr-Labs/eigenlayer-cli/pkg/internal/common"
 	"github.com/Layr-Labs/eigenlayer-cli/pkg/internal/common/flags"
 	"github.com/Layr-Labs/eigenlayer-cli/pkg/telemetry"
+	"github.com/Layr-Labs/eigenlayer-cli/pkg/utils"
+	"github.com/Layr-Labs/eigensdk-go/chainio/clients/elcontracts"
+	"github.com/Layr-Labs/eigensdk-go/logging"
+	eigenSdkUtils "github.com/Layr-Labs/eigensdk-go/utils"
+	gethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
 	"github.com/urfave/cli/v2"
+	"sort"
 )
 
-func ListCmd() *cli.Command {
+type ListUsersReader interface {
+	ListUsers(
+		ctx context.Context,
+		userAddress gethcommon.Address,
+		target gethcommon.Address,
+		selector [4]byte,
+	) ([]gethcommon.Address, error)
+}
+
+func ListCmd(readerGenerator func(logging.Logger, *listUsersConfig) (ListUsersReader, error)) *cli.Command {
 	listCmd := &cli.Command{
 		Name:      "list",
 		Usage:     "user appointee list --account-address <AccountAddress> --target-address <TargetAddress> --selector <Selector>",
 		UsageText: "Lists all appointed users for an account with the provided permissions.",
 		Description: `
 		Lists all appointed users for an account with the provided permissions.
 		`,
-		After: telemetry.AfterRunAction(),
-		Flags: []cli.Flag{
-			&flags.VerboseFlag,
-			&AccountAddressFlag,
+		Action: func(c *cli.Context) error {
+			return listAppointees(c, readerGenerator)
 		},
+		After: telemetry.AfterRunAction(),
+		Flags: listFlags(),
 	}
 
 	return listCmd
 }
+
+func listAppointees(
+	cliCtx *cli.Context,
+	generator func(logging.Logger, *listUsersConfig) (ListUsersReader, error),
+) error {
+	ctx := cliCtx.Context
+	logger := common.GetLogger(cliCtx)
+
+	config, err := readAndValidateListUsersConfig(cliCtx, logger)
+	if err != nil {
+		return eigenSdkUtils.WrapError("failed to read and validate user can call config", err)
+	}
+
```

Comment:
don't we have to set ctx metadata here for network?
`cCtx.App.Metadata["network"] = config.chainID.String()` 
I know it's not ideal. we need something better to have it applied automatically but we can add in all commands right now

Reply:
I'm not clear on why that's needed. Can you explain?

Reply:
https://github.com/Layr-Labs/eigenlayer-cli/blob/master/pkg/telemetry/telemetry.go#L59